aggregator/main.py

```diff
@@ -0,0 +1,47 @@
+import logging
+import os
+
+from flask import Flask, request
+from google.cloud import bigquery
+
+app = Flask(__name__)
+
+
+@app.route('/', methods=['POST'])
+def run_aggregation_queries():
+    """Runs aggregation queries for the given data source and persists
+       results in BigQuery.
+
+       Request form must include the dataset name."""
+    data = request.get_json()
+
+    if data.get('dataset_name') is None:
+        logging.error('Request must include dataset name.')
+        return('', 400)
+
+    dataset_name = data['dataset_name']
+    project_id = os.environ.get('PROJECT_ID')
+    dataset_id = "{}.{}".format(project_id, dataset_name)
+
+    logging.info("Running aggregation queries on dataset: %s", dataset_id)
+    # List and call routines for the dataset. (These have been uploaded to BQ 
+    # through terraform)
+    bq_client = bigquery.Client() 
+    routines = bq_client.list_routines(dataset_id)
+
+    for routine in routines:
```

Comment:
Instead of calling all the routines in a dataset that may or may not be an aggregator, what are your thoughts of prefixing aggregating routines with something like "agg_*" and only executing routine names that match that pattern?

Reply:
Do we expect to have other types of routines? Are the dedupe routines going to be attached to the same datasets?

Reply:
I had thought the dedupe routines were more generic and not tied to specific datasets. But I added a required AGG_ prefix to be safe.

Reply:
Thanks for updating this! From my understanding, BQ routines have to be tied to a specific dataset